docs/source/udf.rst

```diff
@@ -7,6 +7,42 @@ of LiberTEM allows users to run their own reduction functions easily, without ha
 to worry about parallelizing, I/O, the details of buffer management and so on. 
 
 
+How UDF works in layman's terms
+-----------------------------------
+
+.. image:: ./images/diagram.svg
+
+The UDF interface of LiberTEM is heavily utilizing the existing LiberTEM architecture. First,
+data is partitioned into several `"partitions"` and distributed across workers. Then, each partition,
+which can be viewed as a collection of frames, are processed by the user-defined `process_frame` function.
+Here, frames are fetched in an iterative manner, and one could view `process_frame` as `merging` the previous
+computations (i.e., the result computed on previously considered set of frames) and the newly added frame.
+After all the frames in a partition are processed, LiberTEM `merges` the results from each worker, which is called
+by the parameter `merge` in UDF class. Note that in both `process_frame` and `merge`, buffers store the intermediate
+outcomes of the user-defined operations.
+
+Initializing Buffers
+--------------------
+In the UDF interface of LiberTEM, buffers are the tools to save and pass on the
```

Comment:
Maybe this should also mention somewhere the concrete case: that when accessing a buffer via `self.results.buffername`, you get back a view into a numpy array, and that this view will correspond to the current unit of data you are working on (frame/tile/partition/...)